imgproxy: init at 2.8.1

paluh opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

[imgproxy](https://imgproxy.net/) server is not available in nixpkgs. I've packaged only module. I'm not sure if I should also try to provide appropriate service package (this is my first package).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

paluh committed:
imgproxy: init at 2.8.1

erikarvstedt commented:
Please add these fixes: https://github.com/erikarvstedt/nixpkgs/commit/4648cef5b92662669754e3c0da8965f7a5005a2c

paluh commented:
> Welcome to nixpkgs! ￼

Thanks! :-)

I've merged your fixes directly. Do we want to have this merge commit in history or should I squash it into single one and push?

I've rerun `nix-shell -p nix-review --run "nix-review rev HEAD"` and everything seems fine.

erikarvstedt commented:
Please squash it.

paluh commented:
> Please squash it.

No problem. Done :-)
I've rerun the tests and pushed squased version.

P.S. I'm not sure why github claims that I've requested some reviews and incorporates some additional commit in this PR...

paluh commented:
One sec. I've probably listed this additional commit during my rebase. Going to fix it...

paluh commented:
Cleaned up. Sorry for this additional spam :-)

erikarvstedt commented:
This looks great now, congrats! We'll now have to wait for a committer to merge this.

erikarvstedt commented on pkgs/servers/imgproxy/default.nix:
```diff
@@ -0,0 +1,34 @@
+{ lib, buildGoModule, fetchFromGitHub, pkg-config, vips, gobject-introspection }:
+
+buildGoModule rec {
+  version = "2.8.1";
```
Woops, forgot this one: Please move `pname` above `version`.

Infinisil replied:
Fwiw, I've stopped complaining about tiny issues like this, since I've been told there's too much bikeshedding like that, which can be demotivating for contributors.

erikarvstedt commented on pkgs/servers/imgproxy/default.nix:
```diff
@@ -0,0 +1,34 @@
+  pname = "imgproxy";
+
+  src = fetchFromGitHub {
+    owner ="imgproxy";
```
Add a space before `"`

paluh commented:
Fixed and tested. @erikarvstedt thanks a lot for your time, review and support!

erikarvstedt commented on pkgs/servers/imgproxy/default.nix:
```diff
@@ -0,0 +1,34 @@
+
+  goPackagePath = "github.com/imgproxy/imgproxy";
+
+  modSha256 = "0kgd8lwcdns3skvd4bj4z85mq6hkk79mb0zzwky0wqxni8f73s6w";
```
`modSha256` is not needed

erikarvstedt commented on pkgs/servers/imgproxy/default.nix:
```diff
@@ -0,0 +1,34 @@
+    export CGO_LDFLAGS_ALLOW='-(s|w)'
+  '';
+
+  goPackagePath = "github.com/imgproxy/imgproxy";
```
`goPackagePath` can be removed.

erikarvstedt commented on pkgs/servers/imgproxy/default.nix:
```diff
@@ -0,0 +1,34 @@
+
+  goPackagePath = "github.com/imgproxy/imgproxy";
+
+  modSha256 = "0kgd8lwcdns3skvd4bj4z85mq6hkk79mb0zzwky0wqxni8f73s6w";
```
Please put `modSha256` below the `src` definition.

paluh replied:
I've just dropped them and running test build on my machine. Should I bring them back?

erikarvstedt replied:
Sorry, my deleted comment was of course an error, I meant that `goPackagePath` can be removed.

erikarvstedt commented:
Sorry, my go-package-fu is a bit rusty, otherwise I would have included all these changes in my inital patch.

erikarvstedt commented on pkgs/servers/imgproxy/default.nix:
```diff
@@ -0,0 +1,31 @@
+    sha256 = "00hhgh6nrzg2blc6yl8rph5h5w7swlkbh0zgsj7xr0lkm10879pc";
+    rev = "v${version}";
+  };
+  modSha256 = "0kgd8lwcdns3skvd4bj4z85mq6hkk79mb0zzwky0wqxni8f73s6w";
```
Add one newline before `modSha256`

erikarvstedt commented on pkgs/servers/imgproxy/default.nix:
```diff
@@ -0,0 +1,31 @@
+  version = "2.8.1";
+
+  src = fetchFromGitHub {
+    owner = "imgproxy";
```
```
owner = pname;
repo = pname;
```

paluh commented:
> Sorry, my go-package-fu is a bit rusty (...)

No problem. I've removed `goPackagePath` but `modSha256` seems to be required (test build failes with `(..)called without required argument 'modSha256'). I've moved it as you suggested just below `src`.

erikarvstedt commented:
Awesome, it's perfect now. Thanks for you patience! :smiley:

paluh commented:
I've rerun test build. Everything is working fine.

> Awesome, it's perfect now. Thanks for you patience! 

I really appreciate your detailed review. Thanks again!

erikarvstedt commented:
@grahamcofborg build imgproxy

tomberek reviewed (approved):
Builds on NixOS. Functions as expected.

stigtsp reviewed (approved):
Looks good to me

paluh commented:
@erikarvstedt @stigtsp
Do you think that providing `service` expression for this app would be valuable? If so should I include it in this commit or can I do it in a separate one?

erikarvstedt commented:
It's best to do it in a separate PR.

paluh commented:
> It's best to do it in a separate PR.

Cool. I'm going to work on it along with the coding which I do for my internal project deployment.

kalbasit commented on maintainers/maintainer-list.nix:
```diff
@@ -5302,6 +5302,12 @@
+    github = "paluh";
+    githubId = 190249;
+    name = "Tomasz Rybarczyk";
+  };
```
Please put this in a different commit.

kalbasit commented on pkgs/servers/imgproxy/default.nix:
```diff
@@ -0,0 +1,32 @@
+
+  meta = with lib; {
+    description = "Fast and secure on-the-fly image processing server written in Go";
+    homepage = https://imgproxy.net;
```
```suggestion
    homepage = "https://imgproxy.net";
```

paluh committed:
Add paluh to the list of maintainers
